SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -567,7 +567,8 @@ int64 USpatialActorChannel::ReplicateActor()
 		NetDriver->LoadBalanceStrategy != nullptr &&
 		// TODO: the 'bWroteSomethingImportant' check causes problems for actors that need to transition in groups (ex. Character, PlayerController, PlayerState),
 		// so disabling it for now.  Figure out a way to deal with this to recover the perf lost by calling ShouldChangeAuthority() frequently. [UNR-2387]
-		Actor->HasAuthority() &&
+		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID) &&
```

Comment:
Looks like a bug fix to me :)

Reply:
This was a drive-by, I think there can be a mismatch between Role and SpatialOS authority. Either way, authority over the AuthorityIntent component should be the driving check here imo.

Reply:
👍